### PR TITLE
chore(remix-dev): bump esbuild to 0.15.12

### PR DESCRIPTION
**Pull request by mcansh**
Closes: #4154
Closes: #4280
Closes: #4346
Closes: #4639 

- [ ] Docs
- [ ] Tests

Testing Strategy (for esbuild native pnp support): https://github.com/mcansh/remix-pnp-test

testable pre-release `0.0.0-experimental-aed5bfbf7`

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
Once this PR is merged, we can merge #4154 as well

File: packages/create-remix/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "create-remix",
-  "version": "1.7.2",
+  "version": "0.0.0-experimental-aed5bfbf7",
```
Comment (MichaelDeBoey) [inline packages/create-remix/package.json:3]: We should revert this back to `1.7.2` I think?

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: .vscode/deno_resolve_npm_imports.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "// Dependency management is done through `npm` and `node_modules/` instead.": "",
   "// Deno-only dependencies may be imported via URL imports (without using import maps).": "",
   "imports": {
-    "@remix-run/server-runtime": "https://esm.sh/@remix-run/server-runtime@0.0.0-experimental-aed5bfbf7",
+    "@remix-run/server-runtime": "https://esm.sh/@remix-run/server-runtime@1.6.4",
```
Comment (MichaelDeBoey) [inline .vscode/deno_resolve_npm_imports.json:8]: We should bump this to `1.7.2` as well

```suggestion
    "@remix-run/server-runtime": "https://esm.sh/@remix-run/server-runtime@1.7.2",
```

File: templates/deno/.vscode/resolve_npm_imports.json [hunk 3]
```diff
@@ -5,9 +5,9 @@
   "// Deno-only dependencies may be imported via URL imports (without using import maps).": "",
   "imports": {
     "// `@remix-run/deno` code is already a Deno module, so just get types for it directly from `node_modules/`": "",
-    "@remix-run/deno": "https://esm.sh/@remix-run/deno@0.0.0-experimental-aed5bfbf7",
-    "@remix-run/dev/server-build": "https://esm.sh/@remix-run/dev@0.0.0-experimental-aed5bfbf7/server-build",
-    "@remix-run/react": "https://esm.sh/@remix-run/react@0.0.0-experimental-aed5bfbf7",
+    "@remix-run/deno": "../node_modules/@remix-run/deno/index.ts",
+    "@remix-run/dev/server-build": "https://esm.sh/@remix-run/dev@1.6.4/server-build",
+    "@remix-run/react": "https://esm.sh/@remix-run/react@1.6.4",
```
Comment (MichaelDeBoey) [inline templates/deno/.vscode/resolve_npm_imports.json:10]:
```suggestion
    "@remix-run/dev/server-build": "https://esm.sh/@remix-run/dev@1.7.2/server-build",
    "@remix-run/react": "https://esm.sh/@remix-run/react@1.7.2",
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: .vscode/deno_resolve_npm_imports.json (hunk 2 above)
Comment (mcansh) [reply .vscode/deno_resolve_npm_imports.json:8]: yeah these don't get updated during release anymore 🥴

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by MichaelDeBoey** [comment]
@mcansh It appears @pcattori's changes weren't changed to the new `esbuild` version (without the extra Yarn PnP plugin)

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by machour** [comment]
@mcansh Added #4301 to the list of issues that will be fixed by this PR.

**Comment by mcansh** [comment]
> @mcansh Added #4301 to the list of issues that will be fixed by this PR.

that's this PR haha 🙈

**Review by pcattori — APPROVED** [review]

**Pull request merged**